js/preload/preloadMain.js

```diff
@@ -482,6 +482,19 @@ function createAPI() {
         }
     });
 
+    /**
+     * an event triggered by the main process when
+     * the locale is changed
+     * @param dataObj {Object} - locale content
+     */
+    local.ipcRenderer.on('locale-changed', (event, dataObj) => {
+        if (dataObj && typeof dataObj === 'object') {
+            if (dataObj.titleBar) {
```

Comment:
Could combine the two `if` statements.

Reply:
This is a generic event and will be used for more locale contents, so writing a separate if block.